src/panel_gwalker/_gwalker.py

```diff
@@ -146,18 +122,59 @@ def _get_appearance(self, theme):
         config = self._THEME_CONFIG
         return config.get(theme, self.param.appearance.default)
 
+    @param.depends("object")
+    def calculated_fields(self)->list[dict]:
+        """Returns all the fields calculated from the object.
+
+        The calculated fields are a great starting point if you want to customize the fields.
+        """
+        return _raw_fields(self.object)
+
     def _process_param_change(self, params):
-        if self.object is not None and "object" in params:
+        if params.get("object") is not None:
             if not self.fields:
-                params["fields"] = raw_fields(self.object)
+                params["fields"] = self.calculated_fields()
             if not self.config:
                 params["config"] = {}
-        return params
-
-    def _handle_msg(self, msg: any) -> None:
+            if self.server_computation:
+                del params["object"]
+        return super()._process_param_change(params)
+
+    def _compute(self, payload):
+        logger.debug("request: %s", payload)
+        field_specs = self.fields or self.calculated_fields()
```

Comment:
I changed the `field_specs` to also handle situation where user specifies them. Please verify its correct. Thanks.